### PR TITLE
ENH: test `take_along_axis`

### PR DESCRIPTION
**Pull request by ev-br**
Towards gh-299: add testing of `take_along_axis`.

### CODE REVIEW

**Comment by ev-br** [comment]
The spec requires that `axis=-1` has a default value. Pretty much all libraries differ: 

- `jax.numpy` : `axis` is a required argument. jax does not support the 2024.12 revision yet though.
- `numpy`: `axis` is a required argument.
- `cupy`: `axis` is a required argument
- `torch`: `dim=None` defaults to None, but the documentation states that `dim=None` ravels.
- `dask.array`: does not implement `take_along_axis` at all.

https://github.com/data-apis/array-api-compat/pull/317 works around it for numpy and cupy; torch was alread covered; dask.array will need a test skip in https://github.com/data-apis/array-api-compat/pull/317.

cc @crusaderky  for dask

**Comment by ev-br** [comment]
Ran locally the combination of this test and https://github.com/data-apis/array-api-compat/pull/317, for 10_000 examples on each of `torch`, `cupy` and `numpy`.

```
$ ARRAY_API_TESTS_SKIP_DTYPES=uint16,uint32,uint64 ARRAY_API_TESTS_MODULE=array_api_compat.torch pytest array_api_tests/test_indexing_functions.py::test_take_along_axis -v  --max-examples=10_000
```

**Comment by crusaderky** [comment]
Don't expect a dask implementation anytime soon: https://github.com/dask/dask/issues/3663
Best we can do is hack together something in array-api-compat that rechunks to a single chunk, much like it already happens for `sort` and `argsort`. This will work on a "better than nothing" basis, with very big performance caveats.

**Comment by ev-br** [comment]
Thanks, will add a skip with the link to the dask issue then.
